src/proxy/outbound.rs

```diff
@@ -367,6 +367,9 @@ impl OutboundConnection {
                 request_type: RequestType::ToServerWaypoint,
             });
         }
+        if us.workload.gateway_address.is_none() {
```

Comment:
I think this is actually impossible to get with how find_upstream works, which may suggest that we should have a debug_assert or refactor to make the impossible state not representable